### PR TITLE
Fix compilation with GCC7

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/openrct2/game.c [hunk 1]
```diff
@@ -1044,8 +1044,9 @@ void game_autosave()
 	platform_get_time_local(&currentTime);
 
 	utf8 timeName[34];
-	snprintf(timeName, 34, "autosave_%d-%02d-%02d_%02d-%02d-%02d%s",
-		currentDate.year, currentDate.month, currentDate.day, currentTime.hour, currentTime.minute,currentTime.second,
+	snprintf(timeName, 34, "autosave_%04u-%02u-%02u_%02u-%02u-%02u%s",
+		currentDate.year % 10000, currentDate.month % 100, currentDate.day % 100,
+		currentTime.hour % 100, currentTime.minute % 100, currentTime.second % 100,
```
Comment (Gymnasiast) [inline src/openrct2/game.c]: Why not just change timeName[34] to timeName[55]? That gets rid of the warning without this piece of code vomit. Do we really care about 21 bytes during an autosave?

Comment (IntelOrca) [inline src/openrct2/game.c]: This should also use `sizeof(timeName)`

Comment (camthesaxman) [inline src/openrct2/game.c]: I think you should also change the day, month, day_of_week, hour, minute, and second fields in those structs to uint8.

**Comment by janisozaur** [comment]
This should be ready now.

**Comment by Gymnasiast** [comment]
Oh shit, I only now noticed you requested a review from @IntelOrca as well. Sorry about that!

**Comment by IntelOrca** [comment]
The changes are fine.

**Pull request merged**